tests/python/unittest/test_optimizer.py

```diff
@@ -848,6 +848,13 @@ def test_rms():
     wd_options = [{}, {'wd': 0.03}, {'wd': 0.05}, {'wd': 0.07}]
     mp_options = [{}, {'multi_precision': False}, {'multi_precision': True}]
     for dtype in [np.float16, np.float32]:
+        if (dtype == np.float16):
+            rtol=1e-1
+            atol=1e-1
+        else:
+            rtol=1e-2
+            atol=1e-2
```

Comment:
I would prefer:
```Python
rtol = 1e-1 if dtype is np.float16 else 1e-2
atol = 1e-1 if dtype is np.float16 else 1e-2
```

Reply:
IMO the way @frankfliu wrote it is clearer and easier to parse.
But would be great to have a comment explaining it (it was learned by trial and error, right?)

Reply:
done